tests/src/test/java/org/apache/camel/kafkaconnector/services/kafka/StrimziContainer.java

```diff
@@ -0,0 +1,71 @@
+/*
+ *  Licensed to the Apache Software Foundation (ASF) under one or more
+ *  contributor license agreements.  See the NOTICE file distributed with
+ *  this work for additional information regarding copyright ownership.
+ *  The ASF licenses this file to You under the Apache License, Version 2.0
+ *  (the "License"); you may not use this file except in compliance with
+ *  the License.  You may obtain a copy of the License at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ */
+
+package org.apache.camel.kafkaconnector.services.kafka;
+
+import java.util.function.Consumer;
+import com.github.dockerjava.api.command.CreateContainerCmd;
+import org.testcontainers.containers.GenericContainer;
+import org.testcontainers.containers.Network;
+import org.testcontainers.containers.wait.strategy.Wait;
+
+public class StrimziContainer extends GenericContainer {
+    private static final String STRIMZI_CONTAINER = System.getProperty("itest.strimzi.container.image");
+    private static final int KAFKA_PORT = 9092;
+
+    public StrimziContainer(Network network, String name) {
+        super(STRIMZI_CONTAINER);
+
+        withEnv("LOG_DIR", "/tmp/logs");
+        withExposedPorts(KAFKA_PORT);
```

Comment:
I have no idea how it was possible that it was warking without this line with `strimzi/kafka:0.11.4-kafka-2.1.0` image @orpiske ?

Reply:
It works because of the call to [addFixedExposedPort(KAFKA_PORT, KAFKA_PORT)](https://github.com/apache/camel-kafka-connector/pull/66/commits/75e8953c0f3d224d1c1e9309bd3dc7ebd5d24bfc#diff-98b65a9f0c00948336e48a4237b379e5R65) on the start() method. 

Maybe we should add a comment here, but the explanation is that this makes the port mapping from the localhost to inside the container to use the same standard Kafka port (i.e: like running docker -p 9092:9092). The usage of withExposedPort will configure it to use a dynamic port which may require further configuration on the Kafka client. 

The addFixedExposedPort is the default behaviour for the Kafka container, which is why I had originally used it here as well.

Reply:
TBH, I think there's more to it than this. 

The tests work fine if we remove the withExposedPorts(KAFKA_PORT) and run the tests using ```strimzi/kafka:0.11.4-kafka-2.1.0``` image

```mvn -Ditest.zookeeper.container.image=strimzi/kafka:0.11.4-kafka-2.1.0 -Ditest.strimzi.container.image=strimzi/kafka:0.11.4-kafka-2.1.0 -DskipIntegrationTests=false clean verify package```

However, the same thing using ```strimzi/kafka:0.15.0-kafka-2.3.1``` leads to a failure: 

```Caused by: org.testcontainers.containers.ContainerLaunchException: Container startup failed```

Looking at the docker ps output for both, it seems it does not expose the port for the newer image:

```
48ffc42dfae8        strimzi/kafka:0.11.4-kafka-2.1.0    "sh -c 'bin/kafka-se…"   About a minute ago   Up About a minute   0.0.0.0:9092->9092/tcp, 0.0.0.0:32979->9091/tcp, 0.0.0.0:32978->9404/tcp                                                      strimzi
```

```
e30afed7a6bc        strimzi/kafka:0.15.0-kafka-2.3.1    "sh -c 'bin/kafka-se…"   55 seconds ago      Up 54 seconds                                 strimzi
```

Adding the ```withExposedPorts(KAFKA_PORT);``` makes it work for both the older and the newer strimzi images. 

I am guessing that differences on Dockerfiles may also play a part on why it worked without that for the older image.

Reply:
These were my conclusions as well, but it wasn't clear to me looking at the 2 docker files which difference resulted in the different behavior.